docs/testing/unit-testing.md

```diff
@@ -0,0 +1,406 @@
+
+- The Office JavaScript APIs must initialize in a webview control in the context of an Office application (Excel, Word, etc.), so they cannot be loaded in the process in which unit tests run on your development computer.
+- Some unit testing frameworks require that mock objects emulate all the members of the object type that is being mocked, which isn't feasible for the Office JavaScript APIs because some object types have over a hundred members.
+- Unit testing frameworks are stateless, which makes it difficult to test the [application-specific APIs](../develop/understanding-the-javascript-api-for-office.md#api-models) in the Office JavaScript APIs, because they have methods [load()](../develop/application-specific-api-model.md#load) and [sync()](../develop/application-specific-api-model.md#sync) that must be called in a particular order relative to other functions and to each other.
+
+To solve these problems, we have created a library to simplify the creation of mock Office objects in unit tests: [Office-Addin-Mock](https://github.com/OfficeDev/Office-Addin-Scripts/tree/master/packages/office-addin-mock).
+
+The library doesn't depend on the Office JavaScript APIs and it can be used with any JavaScript unit testing framework, including the following among others:
+
+- [Jest](https://jestjs.io)
+- [Mocha](https://mochajs.org/)
+- [Storybook](https://storybook.js.org/docs/react/workflows/unit-testing)
+- [Jasmine](https://jasmine.github.io/)
+
+The examples in this article use the Jest framework. There are examples using the Mocha framework and samples written in TypeScript at [the library's home page](https://github.com/OfficeDev/Office-Addin-Scripts/tree/master/packages/office-addin-mock).
+
+## Prerequisites
+
+This article assumes that you are familiar with the basic concepts of unit testing and mocking, and that you have some experience with a unit testing framework.
+
+> [!TIP]
+> If you are working with Visual Studio, see the article [Unit testing JavaScript and TypeScript in Visual Studio](/visualstudio/javascript/unit-testing-javascript-with-visual-studio.md).
+
+## Install the tool
+
+To install the library open a command prompt, navigate to the root of your add-in project, and then enter the following command:
+
+```command&nbsp;line
+npm install office-addin-mock --save-dev
+```
+
+## Basic usage
+
+1. Require (or import) the library to your test file. The following code is an example:
+
+   ```javascript
+   const OfficeMockObject = require("office-addin-mock");
+   ```
+
+1. Require (or import) the module that contains the add-in function that you want to test. The following is an example that assumes your test file is in a subfolder the folder with your add-in's code files:
```

Comment:
I think you mean " test file is in a subfolder **of** the folder with"

I think it's more common practice to have test files in a sibling folder rather than a sub folder (like source/code.js and test/code.test.js), but I'm not certain of that.